client/src/main/scala/io/delta/sharing/client/DeltaSharingProfileProvider.scala

```diff
@@ -17,22 +17,138 @@
 package io.delta.sharing.client
 
 import java.nio.charset.StandardCharsets.UTF_8
+import java.util.Locale
 
+import com.fasterxml.jackson.core.JsonParser
+import com.fasterxml.jackson.databind.{DeserializationContext, JsonDeserializer}
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize
+import com.fasterxml.jackson.databind.node.ObjectNode
 import org.apache.commons.io.IOUtils
 import org.apache.hadoop.conf.Configuration
 import org.apache.hadoop.fs.Path
 import org.apache.spark.delta.sharing.TableRefreshResult
 
+import io.delta.sharing.client.DeltaSharingProfile.validateNotNullAndEmpty
 import io.delta.sharing.client.util.JsonUtils
 
-case class DeltaSharingProfile(
-    shareCredentialsVersion: Option[Int],
-    endpoint: String = null,
+@JsonDeserialize(using = classOf[DeltaSharingProfileDeserializer])
+sealed trait DeltaSharingProfile {
+  val shareCredentialsVersion: Option[Int]
+  val endpoint: String
+  val `type` : String
+
+  private [client] def validate(): Unit = {
+    if (shareCredentialsVersion.isEmpty) {
+      throw new IllegalArgumentException(
+        "Cannot find the 'shareCredentialsVersion' field in the profile file")
+    }
+
+    if (shareCredentialsVersion.get > DeltaSharingProfile.CURRENT) {
+      throw new IllegalArgumentException(
+        s"'shareCredentialsVersion' in the profile is " +
+          s"${shareCredentialsVersion.get} which is too new. The current release " +
+          s"supports version ${DeltaSharingProfile.CURRENT} and below. Please upgrade to a newer " +
+          s"release.")
+    }
+
+    validateNotNullAndEmpty(endpoint, "endpoint")
+  }
+}
+
+case class BearerTokenDeltaSharingProfile(
+    override val shareCredentialsVersion: Option[Int],
+    override val endpoint: String = null,
     bearerToken: String = null,
     expirationTime: String = null)
+  extends DeltaSharingProfile {
+  override val `type`: String = "bearer_token"
```

Comment:
consider defining these as constants: BEARER_TOKEN

Reply:
sure. added constant for that.